core/src/main/java/org/apache/iceberg/io/DeltaWriter.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.io;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+/**
+ * The write interface could accept INSERT, POS-DELETION, EQUALITY-DELETION. It usually write those operations
+ * in a given partition or bucket.
+ */
+public interface DeltaWriter<T> extends Closeable {
+
+  /**
+   * Write the insert record.
+   */
+  void writeRow(T row) throws IOException;
```

Comment:
I don't think Iceberg classes should throw `IOException` like this. The `FileAppender` interface only throws `IOException` through `Closeable`, which is inherited.